#
# old_revision [3008dd1f70a9e9ddc2bbe30c0fae7aba334836fe]
#
# patch "libpurple/protocols/jabber/roster.c"
#  from [bd15d29a173775903e6619b8bc23311c18f1212c]
#    to [e5258c6a49903c3d2f4a7095d35b863faedff0a4]
#
============================================================
--- libpurple/protocols/jabber/roster.c	bd15d29a173775903e6619b8bc23311c18f1212c
+++ libpurple/protocols/jabber/roster.c	e5258c6a49903c3d2f4a7095d35b863faedff0a4
@@ -274,6 +274,11 @@ static void jabber_roster_update(JabberS
 
 	if(grps) {
 		groups = grps;
+		for (l = groups; l; l = l->next) {
+			purple_debug_info("jabber", "jabber_roster_update(%s): [Source: grps]: groups contains %s",
+							  name, l->data);
+		}
+
 	} else {
 		GSList *buddies = purple_find_buddies(js->gc->account, name);
 		if(!buddies)
@@ -284,8 +289,12 @@ static void jabber_roster_update(JabberS
 			groups = g_slist_append(groups, g->name);
 			buddies = g_slist_remove(buddies, b);
 		}
+		for (l = groups; l; l = l->next) {
+			purple_debug_info("jabber", "jabber_roster_update(%s): [Source: local blist]: groups contains %s",
+							  name, l->data);
+		}
 	}
-
+					  
 	iq = jabber_iq_new_query(js, JABBER_IQ_SET, "jabber:iq:roster");
 
 	query = xmlnode_get_child(iq->node, "query");
@@ -371,6 +382,9 @@ void jabber_roster_alias_change(PurpleCo
 	if(b != NULL) {
 		purple_blist_alias_buddy(b, alias);
 
+		purple_debug_info("jabber", "jabber_roster_alias_change(): Aliased %s to %s",
+				name, alias);
+
 		jabber_roster_update(gc->proto_data, name, NULL);
 	}
 }
@@ -395,6 +409,10 @@ void jabber_roster_group_change(PurpleCo
 			groups = g_slist_append(groups, g->name);
 		buddies = g_slist_remove(buddies, b);
 	}
+
+	purple_debug_info("jabber", "jabber_roster_group_change(): Moving %s from %s to %s",
+				name, old_group, new_group);
+
 	jabber_roster_update(gc->proto_data, name, groups);
 	g_slist_free(groups);
 }
@@ -426,6 +444,9 @@ void jabber_roster_remove_buddy(PurpleCo
 			buddies = g_slist_remove(buddies, tmpbuddy);
 		}
 
+		purple_debug_info("jabber", "jabber_roster_remove_buddy(): Removing %s",
+				buddy->name);
+
 		jabber_roster_update(gc->proto_data, buddy->name, groups);
 		g_slist_free(groups);
 	} else {
